pandas/tests/io/test_pickle.py

```diff
@@ -155,28 +157,44 @@ def test_pickles(current_pickle_data, legacy_pickle):
         compare(current_pickle_data, legacy_pickle, version)
 
 
-def test_round_trip_current(current_pickle_data):
-    def python_pickler(obj, path):
-        with open(path, "wb") as fh:
-            pickle.dump(obj, fh, protocol=-1)
+def python_pickler(obj, path):
+    with open(path, "wb") as fh:
+        pickle.dump(obj, fh, protocol=-1)
 
-    def python_unpickler(path):
-        with open(path, "rb") as fh:
-            fh.seek(0)
-            return pickle.load(fh)
 
+def python_unpickler(path):
+    with open(path, "rb") as fh:
+        fh.seek(0)
+        return pickle.load(fh)
+
+
+named_pickle_writers = {
+    "python": python_pickler,
+    "pandas_proto_default": pd.to_pickle,
+    "pandas_proto_highest": functools.partial(
+        pd.to_pickle, protocol=pickle.HIGHEST_PROTOCOL
+    ),
+    "pandas_proto_4": functools.partial(pd.to_pickle, protocol=4),
+}
+if sys.version_info >= (3, 8):
```

Comment:
import from pandas.compat PY38

Reply:
I assume we should test protocol=5 for all 3.8+, not just `if PY38`?

Reply:
oh sorry, PY38 is defined as >=(3, 8) already